lib/cpflow.rb

```diff
@@ -181,28 +200,33 @@ def self.process_option_params(params)
       hide = command_class::HIDE || deprecated
       with_info_header = command_class::WITH_INFO_HEADER
       validations = command_class::VALIDATIONS
+      subcommand = command_class::SUBCOMMAND
 
       long_description += "\n#{examples}" if examples.length.positive?
 
       # `handle_argument_error` does not exist in the context below,
       # so we store it here to be able to use it
       raise_args_error = ->(*args) { handle_argument_error(commands[name_for_method], ArgumentError, *args) }
 
-      desc(usage, description, hide: hide)
-      long_desc(long_description)
-
-      command_options.each do |option|
-        params = process_option_params(option[:params])
-        method_option(option[:name], **params)
-      end
-
       # We'll handle required options manually in `Config`
       required_options = command_options.select { |option| option[:params][:required] }.map { |option| option[:name] }
       @commands_with_required_options.push(name_for_method.to_sym) if required_options.any?
 
       @commands_with_extra_options.push(name_for_method.to_sym) if accepts_extra_options
 
-      define_method(name_for_method) do |*provided_args| # rubocop:disable Metrics/BlockLength, Metrics/MethodLength
+      klass = subcommand ? klass_for(subcommand) : self
+
+      klass.class_eval do
+        desc(usage, description, hide: hide)
+        long_desc(long_description)
+
+        command_options.each do |option|
+          params = process_option_params(option[:params])
+          method_option(option[:name], **params)
+        end
+      end
+
+      klass.define_method(name_for_method) do |*provided_args| # rubocop:disable Metrics/BlockLength, Metrics/MethodLength
```

Comment:
This is because we need to define it in the subcommand class, correct? There might be some other things we do here that we need to double-check if we also need to do in the subcommand class for them to work.

Reply:
> This is because we need to define it in the subcommand class, correct? 

Yes

> There might be some other things we do here that we need to double-check if we also need to do in the subcommand class for them to work.

Good point, I'll check if I missed something

Reply:
Checked again - didn't find anything new